### PR TITLE
allow some resources to be created while the namespace is terminating

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/3219

@derekwaynecarr ptal

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
I don't like this fix.  

The fix should be to allow RAR through here:
https://github.com/openshift/origin/blob/master/pkg/project/admission/lifecycle/admission.go#L94

**Comment by deads2k** [comment]
Updated to allow the deletion of some resoruces.

File: pkg/cmd/server/origin/reststorage_validation_test.go [hunk 1]
```diff
@@ -65,3 +66,22 @@ func TestAllOpenShiftResourceCoverage(t *testing.T) {
 		t.Errorf("authorizationapi.GroupsToResources[authorizationapi.OpenshiftAllGroupName] is missing %v.  Check pkg/authorization/api/types.go.", strings.ToLower(key))
 	}
 }
+
+// TestCreatesAllowedDuringNamespaceDeletion checks to make sure that the resources in the whitelist are allowed
+func TestCreatesAllowedDuringNamespaceDeletion(t *testing.T) {
```
Comment (derekwaynecarr) [inline pkg/cmd/server/origin/reststorage_validation_test.go]: can we move this test into pkg/project/admission/lifecycle?

File: pkg/project/admission/lifecycle/admission.go [hunk 2]
```diff
@@ -39,14 +41,19 @@ import (
...
 }
 
 type lifecycle struct {
 	client client.Interface
+
+	// creatableResources is a set of resources that can be created even if the namespace is terminating
+	creatableResources util.StringSet
 }
 
+var RecommendedCreatableResources = util.NewStringSet("subjectaccessreviews", "resourceaccessreviews")
```
Comment (derekwaynecarr) [inline pkg/project/admission/lifecycle/admission.go]: can this be hidden?

File: pkg/cmd/server/origin/reststorage_validation_test.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/server/origin/reststorage_validation_test.go]: > can we move this test into pkg/project/admission/lifecycle?

No, that causes a dependency cycle.

Comment (deads2k) [inline pkg/cmd/server/origin/reststorage_validation_test.go]: Actually, this might get included via a different package.  I'll try.

**Comment by deads2k** [comment]
Comments addressed.

**Comment by derekwaynecarr** [comment]
LGTM

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**